### ISSUE TITLE
Health Checks failing due to timeout when executing a long running script step in parallel

### ISSUE THREAD

**Issue by IsaacCalligeros95** [OctopusDeploy/Issues#8563]
### Severity

Low, Workarounds available

### Version

2023.1 -> *

### Latest Version

I could reproduce the problem in the latest build

### What happened?

While a long-running script step is executing health check tasks will start executing but are blocked waiting on the script step by the ScriptIsolationMutex. With the addition of health check timeouts this now results in failed health checks.

### Reproduction

* Configure a 'Run a Script Step' to run on a deployment target, this script will need to run for > 1 minute.
* Trigger a deployment
* Trigger a health check
* Health check fails with a task cancelled exception

### Error and Stacktrace

```shell
Waiting for the script in task ServerTasks-10661 to finish as that script requires that no other Octopus scripts are executing on this target simultaneously. 

The task was cancelled
```

### More Information

This issue depends on a combination of long-running script steps running in parallel with health checks. As this scenario is uncommon (health checks are triggered once daily) this issue is being treated as a low priority. 

### Workaround

* Increase the Machine policy Connect timeout setting.
![image](https://github.com/OctopusDeploy/Issues/assets/101079287/fc221f6f-607c-4705-a10f-435404f97322)
